enterprise/server/remote_execution/runner/runner.go

```diff
@@ -629,6 +591,29 @@ func NewPool(env environment.Env, opts *PoolOptions) (*pool, error) {
 	return p, nil
 }
 
+func (p *pool) initContainerProviders() {
+	providers := make(map[platform.ContainerType]ContainerProvider)
+	dockerProvider, err := docker.NewProvider(p.env, p.imageCacheAuth, p.hostBuildRoot())
+	if err != nil {
+		log.Warningf("Failed to initialize docker container provider: %s", err)
+	}
+	if dockerProvider != nil {
+		providers[platform.DockerContainerType] = dockerProvider
+	}
+	podmanProvider, err := podman.NewProvider(p.env, p.imageCacheAuth, *rootDirectory)
+	if err != nil {
+		log.Warningf("Failed to initialize podman container provider: %s", err)
+	}
+	if podmanProvider != nil {
+		providers[platform.PodmanContainerType] = podmanProvider
+	}
+	providers[platform.FirecrackerContainerType] = firecracker.NewProvider(p.env, p.imageCacheAuth, *rootDirectory)
+	providers[platform.SandboxContainerType] = &sandbox.Provider{}
+	providers[platform.BareContainerType] = &bare.Provider{}
+
+	p.containerProviders = providers
+}
```

Comment:
do you plan on splitting this into OS-specific files in a separate PR? (I think that would let us delete `firecracker_darwin.go` which would be awesome)

Reply:
Yup, it will be another PR. This change was made so that darwin build could be green meanwhile.